lib/publiccloud/instance.pm

```diff
@@ -487,4 +513,176 @@ sub network_speed_test() {
     record_info("curl $rmt_host", $self->run_ssh_command(cmd => "curl -w '$write_out' -o /dev/null -v https://$rmt_host/", proceed_on_failure => 1));
 }
 
+=head2 measure_boottime
+
+    measure_boottime();
+
+Perfomrance measurement of the system Boot time. 
+Mainly used C<systemd-analyze> command for the data extraction.
+Data is then collected in an internal record, ready for storing in a DB.
+Set PUBLIC_CLOUD_PERF_COLLECT true or >0, to activate boottime measurements.
+
+=cut
+
+sub measure_boottime() {
+    my ($self, $instance, $type) = @_;
+    return 0 unless (get_var('PUBLIC_CLOUD_PERF_COLLECT'));
+
+    my $ret = {
+        kernel_release => undef,
+        kernel_version => undef,
+        type => undef,
+        analyze => {},
+        blame => {},
+    };
+
+    record_info("BOOT TIME", 'systemd_analyze');
+    # first deployment analysis
+    my ($systemd_analyze, $systemd_blame) = do_systemd_analyze_time($instance);
+    return 0 unless ($systemd_analyze && $systemd_blame);
+
+    $ret->{analyze}->{$_} = $systemd_analyze->{$_} foreach (keys(%{$systemd_analyze}));
+    $ret->{blame} = $systemd_blame;
+    $ret->{type} = $type;
+    # $ret->{analyze}->{ssh_access} = $startup_time; # placeholder for next implementation
+
+    # Collect kernel version
+    $ret->{kernel_release} = $instance->run_ssh_command(cmd => 'uname -r', proceed_on_failure => 1);
+    $ret->{kernel_version} = $instance->run_ssh_command(cmd => 'uname -v', proceed_on_failure => 1);
+
+    # Do logging to openqa UI
+    $Data::Dumper::Sortkeys = 1;
+    record_info("RESULTS", Dumper($ret));
+    $instance->upload_log("/var/log/" . $_, log_name => 'measure_boottime_' . $_ . '.txt', failok => 1)
+      foreach (qw(cloudregister cloud-init.log cloud-init-output.log messages NetworkManager));
```

Comment:
```suggestion
    my @logfiles = qw(cloudregister cloud-init.log cloud-init-output.log messages NetworkManager);
    $instance->upload_log("/var/log/$_", log_name => 'measure_boottime_$_.txt', failok => 1) foreach (@logfiles);
```

Reply:
I already did so in a previous version of mine, but recently I noted that adding a new code line and a new parameter it didn't provide any improvement in reading and performance, against this way a single line command with the foreach loop seeming much clear with list in it.

Reply:
I'll do that change back.

Reply:
Your change proposal I already did in a previous version (using push -f it isn't visible anymore) but then, I decided to not to use the @log list anymore, leaving the data into the foreach, much more clear.
But now I am going to re-apply it.